diff --git a/CMakeLists.txt b/CMakeLists.txt
index f4407bee0..9df63a563 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -462,7 +462,10 @@ endif()
 #        Finding packages                  #
 ############################################
 
-set(BOOST_COMPONENTS date_time filesystem locale program_options system thread)
+set(BOOST_COMPONENTS date_time filesystem locale program_options thread)
+if(Boost_MAJOR_VERSION EQUAL 1 AND Boost_MINOR_VERSION LESS 69)
+	list(APPEND BOOST_COMPONENTS system)
+endif()
 if(ENABLE_INNOEXTRACT)
 	list(APPEND BOOST_COMPONENTS iostreams)
 endif()
Submodule launcher/lib/innoextract contains modified content
diff --git a/launcher/lib/innoextract/CMakeLists.txt b/launcher/lib/innoextract/CMakeLists.txt
index 491610a..172bd32 100644
--- a/launcher/lib/innoextract/CMakeLists.txt
+++ b/launcher/lib/innoextract/CMakeLists.txt
@@ -160,13 +160,16 @@ else()
 	set(INNOEXTRACT_HAVE_LZMA 0)
 endif()
 
-find_package(Boost REQUIRED COMPONENTS
+set(BOOST_REQUIRED_COMPONENTS
 	iostreams
 	filesystem
 	date_time
-	system
 	program_options
 )
+if(Boost_MAJOR_VERSION EQUAL 1 AND Boost_MINOR_VERSION LESS 69)
+	list(APPEND BOOST_REQUIRED_COMPONENTS system)
+endif()
+find_package(Boost REQUIRED COMPONENTS ${BOOST_REQUIRED_COMPONENTS})
 list(APPEND LIBRARIES ${Boost_LIBRARIES})
 link_directories(${Boost_LIBRARY_DIRS})
 include_directories(SYSTEM ${Boost_INCLUDE_DIR})
